library/src/amd_detail/rocblaslt/src/rocblaslt_auxiliary.cpp

```diff
@@ -2119,35 +2123,55 @@ std::string rocblaslt_internal_get_arch_name()
 
 bool rocblaslt_internal_test_path(const std::string& path)
 {
-#ifdef WIN32
+#ifdef _WIN32
     return ((_access(path.c_str(), 4) != -1) || (_access(path.c_str(), 6) != -1));
 #else
     return access(path.c_str(), R_OK) == 0;
 #endif
 }
 
-#ifndef WIN32
-int hipblaslt_dl_iterate_phdr_callback(struct dl_phdr_info* hdr_info, size_t size, void* data)
+#ifdef _WIN32
+std::string rocblaslt_internal_get_so_path()
 {
-    // uncomment to see all dependent .so files
-    // fprintf(stderr, "hipblaslt so file: %s\n", hdr_info->dlpi_name);
-    std::pair<std::string, std::string>* typedData
-        = reinterpret_cast<std::pair<std::string, std::string>*>(data);
-    if(hdr_info->dlpi_name && strstr(hdr_info->dlpi_name, typedData->second.c_str()))
+    HMODULE hModule = NULL;
+    if(!GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS,
```

Comment:
Is this API from hipruntime? Some APIs in hipruntime have large latencies so may impact the overall performance, but I'm not sure if this one will impact or not.

Reply:
GetModuleHandleEx? That's a Windows API. It's a bit old/gross and when used in this way, it is equivalent to `dladdr`.